front/src/modules/scenario/components/ScenarioExplorer/ScenarioExplorer.tsx

```diff
@@ -73,6 +92,8 @@ const ScenarioExplorer = ({
     return validTrains.length;
   };
 
+  const v2TrainCount = (trainIds: number[]): number => trainIds.length;
```

Comment:
Ts is smart enough to know the type of v2TrainCount :)
```suggestion
  const v2TrainCount = (trainIds: number[]) => trainIds.length;
```